crates/pyrefly_types/src/display.rs

```diff
@@ -604,6 +604,47 @@ impl<'a> ClassDisplayContext<'a> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct HoverCodeSnippet {
+    pub heading: Option<String>,
+    pub body: String,
+    pub default_kind_label: Option<&'static str>,
+}
+
+pub fn format_hover_code_snippet(
```

Comment:
this function doesn't make sense upon reading it with no context here.
- why is it needed when we already have a display for types?
- when should we provide name?
- what is display?
a docstring would help